### DISCUSSION TITLE
Light "Snow Storm" Theme

### DISCUSSION THREAD

**Discussion by tomleo** [arcticicestudio/nord#204] [Suggestions & Ideas]
I would love a light nord colorscheme, something that could be set via `background=light`.

**Comment by arcticicestudio** [comment]
There are proposals to provide light themes for _all_ Nord port projects, but this is not planned in the near future since there are still many open port projects and features that need to be completed before starting with such a big feature. You can try to use auto command groups in your `~/.vimrc` in the meantime to adjust the theme on your own.

```vim
augroup nord-snowstorm
    autocmd!
    autocmd ColorScheme nord highlight <GROUP NAME> ctermfg=<COLOR NUMBER> guifg=<HEX COLOR>
augroup end
```

**Comment by rkiyanchuk** [comment]
Would it require only swap certain colors, or full redefinition of used colors throughout the colorscheme?

**Comment by arcticicestudio** [comment]
@zoresvit Both. All text elements must be adjusted to use a color from the _Polar Night_ palette to be readable (otherwise it's like white-on-white) and UI elements need also be changed to use colors from _Snow Storm_ instead including designs for shading to make elements like the status stand out more.
Most syntax element colors will stay as they are, but punctuations (braces, brackets, colons etc.) must also be adjusted as well as all currently supported plugins that make use of or add UI elements.

[Nord already provides some design guidelines][dcp] and the [website itself is also designed with a light mode][home] that can be toggled to dark with the icon button in the sticky header.

My current scope is to finally complete all implementations for arcticicestudio/nord#94 and then start to [transfer the documentations and assets of all port projects][nd#133] to the new website like [already half finished for Nord itself][n#112].
When all of this is finally done I can start other large epics like this one, but at the moment I have to say that you have to customize Vim on your own with a Nord scoped `augroup` if you'd like to use it in a light mode.

[n#112]: https://github.com/arcticicestudio/nord/issues/112
[nd#133]: https://github.com/arcticicestudio/nord-docs/issues/133
[dcp]: https://www.nordtheme.com/docs/colors-and-palettes
[home]: https://www.nordtheme.com

**Comment by rkiyanchuk** [comment]
@arcticicestudio Sure, I understand, thanks for clarifying 👍🏼

**Comment by casonadams** [comment]
I took a stab at this:
https://github.com/casonadams/nord.vim

It looks okay...  But it is just for vim too. Meh...  For anyone who cares here it is ^^.

**Comment by Hubro** [comment]
Is this still on the roadmap? I have set up my vim config to automatically set `background=light` or `background=dark` based on the time of day. Nord is by far my favorite color scheme, but I can't use it because it doesn't support a light background. Meanwhile I'm using iceberg, which I don't think looks quite as nice :frowning_face:

**Comment by PhilT** [comment]
Not sure if they are planning on this but in the meantime you could try Ayu theme (https://github.com/ayu-theme/ayu-vim). It's got 2 dark and 1 light theme. 

How've you done the switch on dark or light? Is it done on startup of vim?

**Comment by Hubro** [comment]
> How've you done the switch on dark or light? Is it done on startup of vim?

Ok I lied a little bit :stuck_out_tongue: I have a script that runs twice a day that changes a bunch of themes, but I haven't properly integrated Vim yet. For now I have a keybinding set up to toggle between `background=light` and `background=dark`. My plan is to write a plugin that watches a file in my home directory called `.current-global-theme` using inotify and changes the background setting based on the contents. It should be trivial since both vim and neovim supports async IO now, I just haven't looked into it yet.

The idea is that all my applications should change themes automatically without me having to do anything, so it's not good enough to only check on vim startup :slightly_smiling_face:

**Comment by Hubro** [comment]
@PhilT I have now implemented automatic theme switching in neovim using neovim-remote. This is part of my light/dark mode script:

```bash
# Change the theme of running neovim instances using neovim-remote
if which nvr &>/dev/null; then
    ls /tmp/nvim*/0 | while read line; do
        nvr --nostart --servername=$line -c "set background=$1" &
    done
else
    error "Failed to change theme of running nvim instances, please install neovim-remote"
fi
```

This is a lot simpler than my original plan of writing a nvim plugin, and it works perfectly.

This works great with iceberg, but unfortunately it doesn't work at all with ayu, since ayu requires a custom global variable to be changed, then the whole theme to be reloaded manually. I could probably get it to work with an extra autocmd though, I'll probably take a look at that some other day.

**Comment by svengreb** [comment]
Thank you for your patience! 🙏🏼
It‘s been a while [since I had free time to focus more on Nord, and my open source projects in general](https://github.com/orgs/nordtheme/discussions/183#user-content-todays-scope), and invest time in this issue due to [work-life balance](https://github.com/orgs/nordtheme/discussions/183#work-life-balance).

I recently published the first [“Northern Post — The state and roadmap of Nord“ announcement](https://github.com/orgs/nordtheme/discussions/183) which includes all details about the plans and future of the _Nord_ project, including [the goal of catching up with the backlog](https://github.com/orgs/nordtheme/discussions/183#catching-up-on-the-backlog). This issue is part of the backlog and therefore I want to triage and process it to get one step closer to a “clean state“.

Due to the scope of this issue of not being “purely technical“ I‘ll convert it into a discussion for [Nord‘s central community hub on GitHub](https://github.com/orgs/nordtheme/discussions). This way the repository is easier to maintain by separating conversations about the actual code while also making help requests, ideas and other contributions from users more visible and supportable by the community. You can [read and discuss about the goal of reaching the “clean“ contribution triage state in the corresponding announcement](https://github.com/orgs/nordtheme/discussions/202).

This issue is also related to https://github.com/nordtheme/nord/issues/46 which discusses the creation of a kind of “light“ / “day“ variant of Nord that can be rolled out to all ports, including the documentation of the color usage and layout. Therefore the discussion in this thread can be “merged“ and will be included when the time comes.
